openedx/core/djangoapps/content/block_structure/manager.py

```diff
@@ -99,7 +106,15 @@ def get_collected(self):
             BlockStructureTransformers.verify_versions(block_structure)
 
         except (BlockStructureNotFound, TransformerDataIncompatible):
-            block_structure = self._update_collected()
+            if user:
```

Comment:
I think we check if user has `known` attribute and its value is not `False`
```suggestion
            if user and not getattr(user, 'known', False):
```